istioctl/cmd/remove-from-mesh.go

```diff
@@ -201,7 +202,14 @@ func unInjectSideCarFromDeployment(client kubernetes.Interface, deps []appsv1.De
 		depName := strings.Join([]string{dep.Name, dep.Namespace}, ".")
 		sidecarInjected := false
 		podSpec := dep.Spec.Template.Spec.DeepCopy()
-		for _, c := range podSpec.Containers {
+		containers := podSpec.Containers
+		// The sidecar is always named 'istio-proxy', but it isn't a proxy if there is only one container.
+		// (For example, The pod container of ingressgateway is named "istio-proxy", but it isn't a sidecar)
+		if len(containers) == 1 && containers[0].Name == proxyContainerName {
```

Comment:
I think `istio-proxy` will not always be the first container. Can you refer to this PR.
https://github.com/istio/istio/pull/25887/files#diff-beb2c770345282d0846ccdbaab7341d964054df8a3abf696ddee34125b30f86fR193-R202